### PR TITLE
Explain the difference between a user and system session

### PR DESCRIPTION
**Pull request by jelly**
When creating a container with the system session offer a popover with the differences between the owner types.

Closes: #1067

---

image

### CODE REVIEW

File: src/ImageRunModal.jsx [hunk 1]
```diff
@@ -687,7 +687,44 @@ export class ImageRunModal extends React.Component {
...
+                                                              {_("Ports under 1024 can be forwarded")}
+                                                          </TextListItem>
+                                                      </TextList>
+                                                  </TextContent>
+                                                  <TextContent>
+                                                      <Text component={TextVariants.h4}>{cockpit.format("$0 $1", _("User:"), this.props.user)}</Text>
+                                                      <TextList>
+                                                          <TextListItem>
+                                                              {_("Ideal for development")}
+                                                          </TextListItem>
```
Comment (jelly) [inline src/ImageRunModal.jsx:718]: @garrett I can't think of more points for the user session, can you?

File: src/ImageRunModal.jsx [hunk 2]
```diff
@@ -687,7 +687,44 @@ export class ImageRunModal extends React.Component {
...
+                                                          <TextListItem>
+                                                              {_("Resource limits can be set")}
+                                                          </TextListItem>
+                                                          <TextListItem>
+                                                              {_("Checkpoint/restore support")}
+                                                          </TextListItem>
+                                                          <TextListItem>
+                                                              {_("Ports under 1024 can be forwarded")}
+                                                          </TextListItem>
```
Comment (jelly) [inline src/ImageRunModal.jsx:710]: IIRC there are more network related differences but I couldn't find them? IIRC host networking but we don't offer that.

File: src/ImageRunModal.jsx [hunk 3]
```diff
@@ -687,7 +687,44 @@ export class ImageRunModal extends React.Component {
...
+                                                              {_("Ideal for running services")}
+                                                          </TextListItem>
+                                                          <TextListItem>
+                                                              {_("Resource limits can be set")}
+                                                          </TextListItem>
+                                                          <TextListItem>
+                                                              {_("Checkpoint/restore support")}
+                                                          </TextListItem>
+                                                          <TextListItem>
+                                                              {_("Ports under 1024 can be forwarded")}
```
Comment (garrett) [inline src/ImageRunModal.jsx:709]: Can't you use ports under 1024 inside the container and forward to ports 1024 outside? This sounds like the port _inside_ the container must be >= 1024.

Perhaps:

```suggestion
                                                              {_("Ports under 1024 can be mapped")}
```

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (garrett) [reply src/ImageRunModal.jsx:718]: Looks good. I think these points work well.

Are there any network considerations besides the ports?

Out of scope, but interesting and distantly related: 

(The low ports is such is an absurd default policy in this day and age, but that's a different topic and way out of scope — and I think the policy can even be changed manually, but we might even hardcode that somewhere, either in Podman or Cockpit-Podman).

And, someone could remap the port using firewall rules too. (This would actually be a neat feature for Cockpit to handle, probably in the Firewall page, and it would be awesome if we had integration for both Cockpit-Podman and Cockpit-Machines.) But, again, this is out of scope from this PR.

File: src/ImageRunModal.jsx (hunk 3 above)
Comment (jelly) [reply src/ImageRunModal.jsx:709]: Yes, it's only on the outside so podman run -p 22:22 only works as root.

**Comment by jelly** [comment]
Updated the screenshot to reflect the review comments.

**Comment by garrett** [comment]
For rootless containers, we may want to also include something from https://developers.redhat.com/blog/2020/09/25/rootless-containers-with-podman-the-basics#why_rootless_containers_:

Something like this?

"Uses a standard built-in user-based security layer"

(That might be too wordy.)

**Comment by jelly** [comment]
> For rootless containers, we may want to also include something from https://developers.redhat.com/blog/2020/09/25/rootless-containers-with-podman-the-basics#why_rootless_containers_:
> 
> Something like this?
> 
> "Uses the standard user-based security layer"

That sounds a bit vague to me, I changed it to "Running containers as root as an unprivileged user"

**Comment by garrett** [comment]
How about "Restricted by user account permissions"? That's both a pro and con.

**Review by garrett — DISMISSED** [review]

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (martinpitt) [reply src/ImageRunModal.jsx:719]: They are also ideal (or sometimes even required, with e.g. encrypted home dirs) for working with your $HOME data, like handing over your ~/Photos dir as a volume to your image gallery container.

**Review by garrett — APPROVED** [review]

File: src/ImageRunModal.jsx (hunk 2 above)
Comment (garrett) [reply src/ImageRunModal.jsx:711]: It's fine. We have a lot of the most common differences shown already. :+1:
